### PR TITLE
fixed memory leak in pbp ds

### PR DESCRIPTION
**Pull request by M-R-Schaefer**
by reinitializing the processpoolexcecutor every couple of epochs, we avoid a memory leak.
Not entirely sure what causes it in the first place but this is the simplest solution I can think of. 
While reinitializing every epoch is perfectly fine for large datasets, small datasets can get a slight performance hit when restarting the processes all the time.
Since even MP/SPICE can go for more than 10 epochs just fine, I think it's okay to hardcode it to 10.

### CODE REVIEW

File: apax/config/train_config.py [hunk 1]
```diff
@@ -76,10 +76,14 @@ class PBPDatset(DatasetConfig, extra="forbid"):
...
     num_workers : int
         | Number of batches to be processed in parallel.
+    reset_every : int
+        | Number of epochs before reinitializing the ProcessPoolExcecutor.
+        | Avoids memory leaks.
     """
 
     processing: Literal["pbp"] = "pbp"
     num_workers: PositiveInt = 10
+    reset_every: PositiveInt = 10
```
Comment (PythonFZ) [inline apax/config/train_config.py:86]: Can this be set via `dataset:reset_every` or is this really hard coded? Do we know the performance trade-off for setting this larger / smaller?

**Review by PythonFZ — COMMENTED** [review]
Should I test this on SPICE training to be sure it works?

File: apax/config/train_config.py (hunk 1 above)
Comment (M-R-Schaefer) [reply apax/config/train_config.py:86]: it's the config class, this allows you to set it under `dataset:processing:reset_every`

**Comment by M-R-Schaefer** [comment]
> Should I test this on SPICE training to be sure it works?

sounds like a good idea!

**Review by PythonFZ — APPROVED** [review]
Better than expected 👍

**Pull request merged**